### ISSUE TITLE
[raudio] malloc_consolidate() error

### ISSUE THREAD

**Issue by Kapendev** [raysan5/raylib#4262]
### Issue description

While I debug my game, I am unloading a music file very often and I found out that if you do that, then you get this error sometimes:

![Screenshot at 2024-08-17 09-44-49](https://github.com/user-attachments/assets/e3b6ddb7-e90c-4e43-b5ab-554af9e0d700)

I am pretty sure that I am loading the file correctly before I unload it, but I will check again later. I just wanted to make this issue to hear what other people have to say about this. Here is a better look at the call stack:

![Screenshot at 2024-08-17 09-44-23](https://github.com/user-attachments/assets/6bc97ede-07d8-4f89-b4ce-3d97e361a45e)

### Environment

OS: Ubuntu Mate 24.04 LTS

**Comment by Peter0x44** [comment]
Can you provide code which reproduces this?

**Comment by Kapendev** [comment]
Here @Peter0x44  
This is written in DLang, but it should be very easy to translate it to C.
Let it run and it will show the error after some time.

```d
import popka.ray;

Music music;

void main() {
    InitWindow(800, 450, "raylib");
    InitAudioDevice();

    music = LoadMusicStream("song.mp3");
    PlayMusicStream(music);

    while (!WindowShouldClose()) {
        BeginDrawing();
        if (true) {
            UnloadMusicStream(music);
            music = LoadMusicStream("song.mp3");
            PlayMusicStream(music);
        }
        ClearBackground(RAYWHITE);
        UpdateMusicStream(music);
        EndDrawing();
    }

    UnloadMusicStream(music);
    CloseAudioDevice();
    CloseWindow();
}
```

**Comment by Kapendev** [comment]
You might also need to rerun it, because sometimes nothing happens.

**Comment by raysan5** [comment]
@Kapendev I'm afraid this issue could be related to DLang, I can't reproduce it in C.

**Comment by Kapendev** [comment]
Well, I compiled the same program with C and I get the same error. I think it might be my setup then. No idea.

**Comment by Kapendev** [comment]
I discovered that the issue might actually be related to my setup. I compiled raylib into a static library and linked it to my test program, instead of using the official one provided on GitHub. After doing this, the error no longer occurs. I tested it with both C and D to confirm, and the results were consistent. No errors with my version, but the official library still triggers the error. For context, I'm using Ubuntu Mate 24.04 LTS.

@raysan5 Sorry to ping you, but could this issue possibly be related to glibc?
